adapters/rubicon/rubicon_test.go

```diff
@@ -572,52 +574,105 @@ func TestResolveVideoSizeId(t *testing.T) {
 	}
 }
 
-func TestResolveBidFloorAttributes(t *testing.T) {
+func TestOpenRTBRequestWithDifferentBidFloorAttributes(t *testing.T) {
 	testScenarios := []struct {
-		bidFloor            float64
-		bidFloorCur         string
-		expectedBidFloor    float64
-		expectedBidFloorCur string
+		bidFloor         float64
+		bidFloorCur      string
+		setMock          func(m *mock.Mock)
+		expectedBidFloor float64
+		expectedErrors   []error
 	}{
 		{
-			bidFloor:            1,
-			bidFloorCur:         "EUR",
-			expectedBidFloor:    1.2,
-			expectedBidFloorCur: "USD",
-		},
-		{
-			bidFloor:            1,
-			bidFloorCur:         "Eur",
-			expectedBidFloor:    1.2,
-			expectedBidFloorCur: "USD",
-		},
-		{
-			bidFloor:            0,
-			bidFloorCur:         "EUR",
-			expectedBidFloor:    0,
-			expectedBidFloorCur: "EUR",
+			bidFloor:         1,
+			bidFloorCur:      "WRONG",
+			setMock:          func(m *mock.Mock) { m.On("GetRate", "WRONG", "USD").Return(2.5, errors.New("some error")) },
+			expectedBidFloor: 0,
+			expectedErrors: []error{
+				&errortypes.BadInput{Message: "Unable to convert provided bid floor currency from WRONG to USD"},
+			},
 		},
 		{
-			bidFloor:            -1,
-			bidFloorCur:         "EUR",
-			expectedBidFloor:    -1,
-			expectedBidFloorCur: "EUR",
+			bidFloor:         1,
+			bidFloorCur:      "USD",
+			setMock:          func(m *mock.Mock) {},
+			expectedBidFloor: 1,
+			expectedErrors:   nil,
 		},
 		{
-			bidFloor:            1,
-			bidFloorCur:         "USD",
-			expectedBidFloor:    1,
-			expectedBidFloorCur: "USD",
+			bidFloor:         1,
+			bidFloorCur:      "EUR",
+			setMock:          func(m *mock.Mock) { m.On("GetRate", "EUR", "USD").Return(1.2, nil) },
+			expectedBidFloor: 1.2,
+			expectedErrors:   nil,
 		},
 	}
 
 	for _, scenario := range testScenarios {
-		bidFloor, bidFloorCur := resolveBidFloorAttributes(scenario.bidFloor, scenario.bidFloorCur)
-		assert.Equal(t, scenario.expectedBidFloor, bidFloor)
-		assert.Equal(t, scenario.expectedBidFloorCur, bidFloorCur)
+		mockConversions := &mockCurrencyConversion{}
+		scenario.setMock(&mockConversions.Mock)
+
+		extraRequestInfo := adapters.ExtraRequestInfo{
+			CurrencyConversions: mockConversions,
```

Comment:
Is this the only reason why you changed `CurrencyConversions ` to public?

Reply:
Actually yes, it there is any better way, for my purpose? Or everything is fine?

Reply:
It is fine @snahornyi in the future we plan to improve the JSON testing framework. We also encourage tests to be written in JSON instead of coded in Golang but, for the moment we agree with this approach